catalogs/catalog-hadoop/src/test/java/com/datastrato/gravitino/catalog/hadoop/integration/test/HDFSKerberosIT.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.catalog.hadoop.integration.test;
+
+import com.datastrato.gravitino.integration.test.container.ContainerSuite;
+import com.datastrato.gravitino.integration.test.container.HiveContainer;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.PrivilegedAction;
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.security.AccessControlException;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag("gravitino-docker-it")
+public class HDFSKerberosIT {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSKerberosIT.class);
+
+  private static final ContainerSuite containerSuite = ContainerSuite.getInstance();
+  private static final String CLIENT_PRINCIPAL = "cli@HADOOPKRB";
+  private static UserGroupInformation clientUGI;
+
+  @BeforeAll
+  public static void setup() throws IOException {
+    containerSuite.startKerberosHiveContainer();
+
+    // Copy the keytab and krb5.conf from the container
+    containerSuite
+        .getKerberosHiveContainer()
+        .getContainer()
+        .copyFileFromContainer("/etc/admin.keytab", "/tmp/client.keytab");
```

Comment:
It would be better to make this file a temp file with random prefix, also make sure to remove this when UT is finished.